datafusion/common/src/config.rs

```diff
@@ -280,6 +280,10 @@ config_namespace! {
         /// using the provided `target_partitions` level
         pub repartition_joins: bool, default = true
 
+        /// Should DataFusion allow symmetric hash joins for unbounded data sources even when
+        /// its inputs do not have any ordering or filtering
+        pub allow_symmetric_joins_without_pruning: bool, default = true
```

Comment:
I don't understand how a symmetric hash join could generate correct results when the inputs don't have any ordering 🤔  Maybe we can add some additional comments about under what circumstances one would enable 
 / disable this option.

Reply:
SHJ will always produce correct results, but it will use twice as much memory (assuming inputs are of the same size) for no gain except pipelining.

Some more explanation about this option: It is not always possible to detect 100% accurately whether pruning may occur or not -- the system may think pruning is not possible where it is actually possible. Therefore, one would enable this option if they have a-priori knowledge that data would indeed lend itself to pruning.

Reply:
Thank you -- this explanation and the updated comments help to clarify